builder_test.go

```diff
@@ -775,3 +775,43 @@ func TestBuilder(t *testing.T) {
 		})
 	}
 }
+
+func TestRunWithEnvArgConflict(t *testing.T) {
+	f, err := os.Open("dockerclient/testdata/Dockerfile.envargconflict")
+	if err != nil {
+		t.Fatal(err)
+	}
+	node, err := ParseDockerfile(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	b := NewBuilder(nil)
+	from, err := b.From(node)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if from != "ubuntu:18.04" {
+		t.Fatalf("unexpected from: %s", from)
+	}
+	for _, child := range node.Children {
+		step := b.Step()
+		if err := step.Resolve(child); err != nil {
+			t.Fatal(err)
+		}
+		if err := b.Run(step, LogExecutor, false); err != nil {
+			t.Fatal(err)
+		}
+	}
+	configString := fmt.Sprintf("%v", b.Config())
+	expectedValue := "USER_NAME=my_user_env"
+	if !strings.Contains(configString, expectedValue) {
```

Comment:
Maybe add a test to check `my_user_arg` value does not contain?

Reply:
Good thought @QiWang19 .  Added and pushed.